### PR TITLE
[Connection pool] Return boolean from "validate" method

### PR DESCRIPTION
**Pull request by yan-kunitski**
According to `generic-pool` [docs](https://github.com/coopernurse/node-pool#creating-a-pool) the validate method should always return `Promise<boolean>`. Otherwise, it may lead to unforeseen consequences. For instance, if `heartbeatAsync()` throws `Could not reach Snowflake` then `unhandledRejection` event pops up (node process dies 💀 if you use the latest versions).

### CODE REVIEW

**Comment by yan-kunitski** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by yan-kunitski** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/core.js [hunk 1]
```diff
@@ -278,8 +278,18 @@ function Core(options)
...
-      return ((heartbeat[0][1] == 1) && connection.isUp());
+      try
+      {
+        var heartbeat = await connection.heartbeatAsync();
+
+        return ((heartbeat[0][1] == 1) && connection.isUp());
+      }
+      catch (err)
+      {
+        console.log('Unbale to validate: ' + err.message);
```
Comment (Luffi2539) [inline lib/core.js:289]: typo ('Unable')

**Comment by sfc-gh-dprzybysz** [comment]
It was fixed in https://github.com/snowflakedb/snowflake-connector-nodejs/pull/523 and https://github.com/snowflakedb/snowflake-connector-nodejs/pull/525

**Pull request closed without merging**